test/functional/test_framework/test_node.py

```diff
@@ -187,7 +187,11 @@ def wait_for_rpc_connection(self):
                 if e.errno != errno.ECONNREFUSED:  # Port not yet open?
                     raise  # unknown IO error
             except JSONRPCException as e:  # Initialization phase
-                if e.error['code'] != -28:  # RPC in warmup?
+                if e.error['code'] == -28:  # RPC in warmup?
+                    pass
+                elif e.error['code'] == -342: # RPC handler unregistered, but http server haven't been stopped yet.
```

Comment:
I think this comment is confusing without mentioning why the server is shutting down.  Maybe change it to something like:

```python
# Service unavailable, RPC server started but is shutting down due to error
```

Reply:
Comment added